src/twisted/internet/test/test_inlinecb.py

```diff
@@ -128,20 +128,61 @@ class ForwardTraceBackTests(SynchronousTestCase):
 
     def test_forwardTracebacks(self):
         """
-        Test that chained inlineCallbacks are properly forwarding the traceback informations
-        from generator to generator
-        Reproducing #9175
+        Chained inlineCallbacks are forwarding the traceback informations
+        from generator to generator.
+
+        A first simple test with a couple of inline callbacks.
         """
 
         @inlineCallbacks
         def erroring():
-            raise Exception()
             yield "forcing generator"
+            raise Exception('Error Marker')
 
         @inlineCallbacks
         def calling():
             yield erroring()
 
         d = calling()
         f = self.failureResultOf(d)
-        self.assertIn("in erroring", f.getTraceback())
+        tb = f.getTraceback()
+        self.assertIn("in erroring", tb)
+        self.assertIn("Error Marker", tb)
+
+
+    def test_forwardLotsOfTracebacks(self):
+        """
+        Several Chained inlineCallbacks gives information about all generators.
+
+        A wider test with a 4 chained inline callbacks.
```

Comment:
what is the difference between 2 chained callbacks and 4 chained?

Do we need both tests?

Reply:
I agree we don't really need two tests, but the current code passes test_forwardTracebacks, but not test_forwardLotsOfTracebacks.

So if there is a regression, this adds the info whether this is in the basic traceback generation, and/or the "advanced" stack traversal code.

Then I like to have a simple basic test, and then more tests which catch corner cases that we may find later (hence the ref to bugs). This is a methodology that we use at Buildbot, I'm perfectly okay to follow the one of Twisted, and to remove the first test, if you insist.

Reply:
As you are explaining it now, we do need 2 tests:)

Please add all this information in the docstring, so that in 2 years from now, when someone will look at these tests will know why there are 2 tests without checking the whole PR and the history of those changes.

Thanks!

Reply:
Right. I tried to add more details in docstrings, and in failure.py's comments